### DISCUSSION TITLE
NotNullViolation if any column (not just the primary_key) of related table is None.

### DISCUSSION THREAD

**Discussion by jalvespinto** [sqlalchemy/sqlalchemy#7630] [Usage Questions] [answered]
I have the two tables below. I am trying to persist a nutri_facts row that have 'trans_fat = None', but then I get a NotNullViaolation. Shouldn't this error come up only if 'nutri_facts.id' is null? 
```

products = Table(
    'products', mapper_registry.metadata,
    Column('id', String(255), primary_key=True, unique=True),
    Column('name', String(255)),
    Column('brand', String(255)),
    Column('category', String(255)),
    Column('barcode', String(255)),
    Column('nutri_facts_id', ForeignKey('nutri_facts.id')),
    Column('ingredients', Text),
)

nutri_facts = Table(
    'nutri_facts', mapper_registry.metadata,
    Column('id', Integer, primary_key=True, autoincrement=True),
    Column('calories', Float(precision=7)),
    Column('protein', Float(precision=7)),
    Column('carbohydrate', Float(precision=7)),
    Column('total_fat', Float(precision=7)),
    Column('saturated_fat', Float(precision=7)),
    Column('trans_fat', Float(precision=7)),
    Column('dietary_fiber', Float(precision=7)),
    Column('sodium', Float(precision=7)),
)
```

**Comment by zzzeek** [comment, marked as answer]
> I have the two tables below. I am trying to persist a nutri_facts row that have 'trans_fat = None', but then I get a NotNullViaolation. Shouldn't this error come up only if 'nutri_facts.id' is null?

this would be correct, if your actual database has a schema which matches the declarations in this metadata.  however there's no such guarantee of that unless this specific metadata were used to create the tables.  if the tables are pre-existing, there could be a NOT NULL constraint on that column in the database.
